// @angular-material-extensions/jumbotron/bundles/jumbotron.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/material'), require('@angular/flex-layout')) :
    typeof define === 'function' && define.amd ? define(['exports', '@angular/common', '@angular/core', '@angular/material', '@angular/flex-layout'], factory) :
    (global = global || self, factory(global.angularMaterialExtensionsJumbotron = {}, global.ng.common, global.ng.core, global.material, global.flexLayout));
}(this, function (exports, common, core, material, flexLayout) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MatJumbotronContentComponent = /** @class */ (function () {
        function MatJumbotronContentComponent() {
        }
        /**
         * @return {?}
         */
        MatJumbotronContentComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        MatJumbotronContentComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'mat-jumbotron-content',
                        template: "\n    <ng-template>\n      <ng-content></ng-content>\n    </ng-template>\n  ",
                        styles: ["\n    :host{display:block}\n  "]
                    },] },
        ];
        /** @nocollapse */
        MatJumbotronContentComponent.ctorParameters = function () { return []; };
        MatJumbotronContentComponent.propDecorators = {
            content: [{ type: core.ViewChild, args: [core.TemplateRef, { static: false },] }]
        };
        return MatJumbotronContentComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MatJumbotronComponent = /** @class */ (function () {
        function MatJumbotronComponent() {
        }
        MatJumbotronComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'mat-jumbotron',
                        template: "\n    <mat-card fxLayout=\"row\" fxLayoutAlign=\"center center\">\n\n      <mat-card-content fxLayout=\"row\"\n                        fxFlex\n                        fxLayout.xs=\"column\"\n                        fxLayoutGap=\"2rem\"\n                        fxLayoutAlign=\"center center\">\n        <!--main image-->\n        <img *ngIf=\"imgURL\"\n             mat-card-xl-image\n             [src]=\"imgURL\">\n\n        <!--content-->\n        <div fxLayout=\"column\" [fxFlex]=\"imgURL ? 50 : 100\">\n          <ng-container [ngTemplateOutlet]=\"jumbotronContent.content\"></ng-container>\n        </div>\n      </mat-card-content>\n\n    </mat-card>\n  ",
                        styles: ["\n    :host{display:block}:host ::ng-deep mat-card{background:#e9ecef!important}:host ::ng-deep mat-card img{-o-object-fit:cover;object-fit:cover}:host ::ng-deep mat-card h1{font-weight:400!important}:host ::ng-deep mat-card p{margin-top:0;margin-bottom:1rem;font-size:1.4rem}\n  "],
                        encapsulation: core.ViewEncapsulation.Emulated
                    },] },
        ];
        /** @nocollapse */
        MatJumbotronComponent.ctorParameters = function () { return []; };
        MatJumbotronComponent.propDecorators = {
            jumbotronContent: [{ type: core.ContentChild, args: [MatJumbotronContentComponent, { static: false },] }],
            imgURL: [{ type: core.Input }]
        };
        return MatJumbotronComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MatJumbotronModule = /** @class */ (function () {
        function MatJumbotronModule() {
        }
        MatJumbotronModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            flexLayout.FlexLayoutModule,
                            material.MatCardModule,
                        ],
                        exports: [MatJumbotronComponent, MatJumbotronContentComponent],
                        declarations: [MatJumbotronComponent, MatJumbotronContentComponent]
                    },] },
        ];
        return MatJumbotronModule;
    }());

    exports.MatJumbotronComponent = MatJumbotronComponent;
    exports.MatJumbotronContentComponent = MatJumbotronContentComponent;
    exports.MatJumbotronModule = MatJumbotronModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
