// @angular-material-extensions/jumbotron/esm5/jumbotron.es5.js
import { CommonModule } from '@angular/common';
import { Component, ViewChild, TemplateRef, ViewEncapsulation, ContentChild, Input, NgModule } from '@angular/core';
import { MatCardModule } from '@angular/material';
import { FlexLayoutModule } from '@angular/flex-layout';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MatJumbotronContentComponent = /** @class */ (function () {
    function MatJumbotronContentComponent() {
    }
    /**
     * @return {?}
     */
    MatJumbotronContentComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    MatJumbotronContentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mat-jumbotron-content',
                    template: "\n    <ng-template>\n      <ng-content></ng-content>\n    </ng-template>\n  ",
                    styles: ["\n    :host{display:block}\n  "]
                },] },
    ];
    /** @nocollapse */
    MatJumbotronContentComponent.ctorParameters = function () { return []; };
    MatJumbotronContentComponent.propDecorators = {
        content: [{ type: ViewChild, args: [TemplateRef, { static: false },] }]
    };
    return MatJumbotronContentComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MatJumbotronComponent = /** @class */ (function () {
    function MatJumbotronComponent() {
    }
    MatJumbotronComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mat-jumbotron',
                    template: "\n    <mat-card fxLayout=\"row\" fxLayoutAlign=\"center center\">\n\n      <mat-card-content fxLayout=\"row\"\n                        fxFlex\n                        fxLayout.xs=\"column\"\n                        fxLayoutGap=\"2rem\"\n                        fxLayoutAlign=\"center center\">\n        <!--main image-->\n        <img *ngIf=\"imgURL\"\n             mat-card-xl-image\n             [src]=\"imgURL\">\n\n        <!--content-->\n        <div fxLayout=\"column\" [fxFlex]=\"imgURL ? 50 : 100\">\n          <ng-container [ngTemplateOutlet]=\"jumbotronContent.content\"></ng-container>\n        </div>\n      </mat-card-content>\n\n    </mat-card>\n  ",
                    styles: ["\n    :host{display:block}:host ::ng-deep mat-card{background:#e9ecef!important}:host ::ng-deep mat-card img{-o-object-fit:cover;object-fit:cover}:host ::ng-deep mat-card h1{font-weight:400!important}:host ::ng-deep mat-card p{margin-top:0;margin-bottom:1rem;font-size:1.4rem}\n  "],
                    encapsulation: ViewEncapsulation.Emulated
                },] },
    ];
    /** @nocollapse */
    MatJumbotronComponent.ctorParameters = function () { return []; };
    MatJumbotronComponent.propDecorators = {
        jumbotronContent: [{ type: ContentChild, args: [MatJumbotronContentComponent, { static: false },] }],
        imgURL: [{ type: Input }]
    };
    return MatJumbotronComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MatJumbotronModule = /** @class */ (function () {
    function MatJumbotronModule() {
    }
    MatJumbotronModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        FlexLayoutModule,
                        MatCardModule,
                    ],
                    exports: [MatJumbotronComponent, MatJumbotronContentComponent],
                    declarations: [MatJumbotronComponent, MatJumbotronContentComponent]
                },] },
    ];
    return MatJumbotronModule;
}());

export { MatJumbotronComponent, MatJumbotronContentComponent, MatJumbotronModule };
